### PR TITLE
[wip] feat: generalize lighthouse

### PR DESCRIPTION
**Pull request by aaarichter**
- [x] add `runLighthouse` to run multiple categories ('performance', 'accessibility', 'best-practices', 'seo', 'pwa')
- [x] keep `runLighthouseAudit` which previously ran only accessiblity audits
- [x] upgrade packages

### CODE REVIEW

File: lib/browser/lighthouse.js [hunk 1]
```diff
@@ -34,6 +34,7 @@
 
 const assert = require('assertive');
 const isEmpty = require('lodash/isEmpty');
+const deepMerge = require('deepmerge');
```
Comment (dbushong) [inline lib/browser/lighthouse.js]: does this do something for us that `lodash/merge` doesn't?

Comment (aaarichter) [reply lib/browser/lighthouse.js]: thanks @dbushong . let me look into it

File: lib/browser/lighthouse.js [hunk 2]
```diff
@@ -42,59 +43,52 @@ function getTotalScore(success, failure) {
...
 
+/**
+ *
+ * @return {Object}
+ * @param config
+ */
+function getAccessibilityConfig(config) {
+  if (!config) {
+    config = Object.assign({}, defaultConfig);
+    config.settings.onlyCategories.push('accessibility');
```
Comment (dbushong) [inline lib/browser/lighthouse.js]: This still modifies `defaultConfig`, which seems bad.  Options include:

```js
config = JSON.parse(JSON.stringify(defaultConfig));
```

or

```js
const cloneDeep = require('lodash/cloneDeep');
// ...
config = cloneDeep(defaultConfig);
```

or

```js
config = Object.assign({}, config, {
  settings: Object.assign({}, config.settings, {
    onlyCategories: config.settings.onlyCategories.concat(['accessibility']),
  }),
});
```

File: lib/browser/lighthouse.js [hunk 3]
```diff
@@ -163,9 +159,50 @@ exports.assertLighthouseScore = function assertLighthouseScore(
...
-exports.runLighthouseAudit = function runLighthouseAudit(flags, config) {
-  return this.getLighthouseData(flags, config || lhConfig).then(parseLhResult);
+/**
+ *
+ * @param {string|Array<string>} categories - LH categories: 'performance', 'accessibility', 'best-practices', 'seo', 'pwa'
+ * @param {Object=} lhConfig - Custom Lighthouse config
+ */
+exports.runLighthouse = function runLighthouse(categories = [], lhConfig = {}) {
+  categories = !Array.isArray(categories) ? [categories] : categories;
+  let config = Object.assign({}, defaultConfig);
```
Comment (dbushong) [inline lib/browser/lighthouse.js]: similar problem as above; you could either do a deep clone here, or in all of the merges below, do them like: `config = merge({}, config, catConfig)`

File: lib/browser/lighthouse.js (hunk 2 above)
Comment (aaarichter) [reply lib/browser/lighthouse.js]: addressing

**Comment by aaarichter** [comment]
closing in favor of #50

**Pull request closed without merging**